crates/sui-core/src/authority.rs

```diff
@@ -4148,6 +4148,22 @@ impl AuthorityState {
         Some(tx)
     }
 
+    #[instrument(level = "debug", skip_all)]
+    fn create_coin_deny_list_state_tx(
+        &self,
+        epoch_store: &Arc<AuthorityPerEpochStore>,
+    ) -> Option<EndOfEpochTransactionKind> {
+        if !epoch_store.protocol_config().enable_coin_deny_list() {
+            return None;
+        }
+
+        if epoch_store.coin_deny_list_state_exists() {
+            return None;
+        }
+
+        Some(EndOfEpochTransactionKind::CoinDenyListStateCreate)
```

Comment:
again here I think it keeps the code easier to read if you follow the existing pattern? the existing pattern may be dumb but IMHO it's better to be consistent, and if you want to change it then change them all together